dockstore-webservice/src/test/java/io/swagger/api/impl/ToolsImplCommonTest.java

```diff
@@ -0,0 +1,155 @@
+        DockstoreWebserviceConfiguration actualConfig = new DockstoreWebserviceConfiguration();
+        actualConfig.setHostname("localhost");
+        actualConfig.setPort("8080");
+        actualConfig.setScheme("http");
+        Pair<Tool, Table<String, SourceFile.FileType, Object>> toolTablePair = ToolsImplCommon.convertEntryToTool(workflow, actualConfig);
+        Tool actualTool = toolTablePair.getLeft();
+        ToolVersion expectedToolVersion = new ToolVersion();
+        expectedToolVersion.setName("master");
+        expectedToolVersion
+                .setUrl("http://localhost:8080/api/ga4gh/v2/tools/%23workflow%2Fgithub.com%2FICGC-TCGA-PanCancer%2Fwdl-pcawg-sanger-cgp-workflow/versions/master");
+        expectedToolVersion.setId("#workflow/github.com/ICGC-TCGA-PanCancer/wdl-pcawg-sanger-cgp-workflow:master");
+        List<ToolVersion.DescriptorTypeEnum> descriptorTypeList = new ArrayList<>();
+        descriptorTypeList.add(ToolVersion.DescriptorTypeEnum.WDL);
+        expectedToolVersion.setImage("");
+        expectedToolVersion.setDescriptorType(descriptorTypeList);
+        expectedToolVersion.setDockerfile(false);
+        // Meta-version dates are currently dependant on the environment, disabling for now
+        expectedToolVersion.setMetaVersion(null);
+        expectedToolVersion.setVerified(false);
+        expectedToolVersion.setVerifiedSource("");
+        List<ToolVersion> expectedToolVersions = new ArrayList<>();
+        expectedToolVersions.add(expectedToolVersion);
+        Tool expectedTool = new Tool();
+        expectedTool
+                .setUrl("http://localhost:8080/api/ga4gh/v2/tools/%23workflow%2Fgithub.com%2FICGC-TCGA-PanCancer%2Fwdl-pcawg-sanger-cgp-workflow");
+        expectedTool.setId("#workflow/github.com/ICGC-TCGA-PanCancer/wdl-pcawg-sanger-cgp-workflow");
+        expectedTool.setOrganization("ICGC-TCGA-PanCancer");
+        expectedTool.setToolname("wdl-pcawg-sanger-cgp-workflow");
+        expectedTool.setToolclass(ToolClassesApiServiceImpl.getWorkflowClass());
+        expectedTool.setDescription("");
+        expectedTool.setAuthor("Unknown author");
+        // Meta-version dates are currently dependant on the environment, disabling for now
+        expectedTool.setMetaVersion(null);
+        expectedTool.setContains(Collections.EMPTY_LIST);
+        expectedTool.setVerified(false);
+        expectedTool.setVerifiedSource("[]");
+        expectedTool.setSigned(false);
+        expectedTool.setVersions(expectedToolVersions);
+        actualTool.setMetaVersion(null);
+
+        // Meta-version dates are currently dependant on the environment, disabling for now
+        List<ToolVersion> versions = actualTool.getVersions();
+        versions.stream().forEach(version -> version.setMetaVersion(null));
+        actualTool.setVersions(versions);
+
+        Assert.assertEquals(actualTool, expectedTool);
```

Comment:
A good start, but let's also check a few more things. 
i.e. like verified tools vs their versions are consistent, verified sources are dealth with properly
Was there a different PR where you tested for collisions in this API? etc.